Scripts/Runtime/Entities/TaskSystem/TaskData/DataStream/Abstract/AbstractArrayDataStream.cs

```diff
@@ -0,0 +1,142 @@
+
+        public DeferredNativeArrayScheduleInfo ScheduleInfo
+        {
+            get => Live.ScheduleInfo;
+        }
+
+        protected AbstractArrayDataStream(AbstractTaskDriver taskDriver, AbstractTaskSystem taskSystem) : base(taskDriver, taskSystem)
+        {
+            Live = new DeferredNativeArray<T>(Allocator.Persistent,
+                                              Allocator.TempJob);
+        }
+
+        protected override void DisposeDataStream()
+        {
+            Live.Dispose();
+            base.DisposeDataStream();
+        }
+        
+        //*************************************************************************************************************
+        // SERIALIZATION
+        //*************************************************************************************************************
+
+        //TODO: #83 - Add support for Serialization. Hopefully from the outside or via extension methods instead of functions
+        //here but keeping the TODO for future reminder.
+
+        //*************************************************************************************************************
+        // CONSOLIDATION
+        //*************************************************************************************************************
+        
+        protected override JobHandle ConsolidateForFrame(JobHandle dependsOn)
+        {
+            dependsOn = JobHandle.CombineDependencies(dependsOn,
+                                                      AccessController.AcquireAsync(AccessType.ExclusiveWrite));
+
+
+            ConsolidateArrayDataStreamJob consolidateArrayDataStreamJob = new ConsolidateArrayDataStreamJob(Pending,
+                                                                                                            Live
+#if DEBUG
+                                                                                                           ,
+                                                                                                            Debug_ProfilerMarker
+#endif
+#if ANVIL_DEBUG_LOGGING_EXPENSIVE
+                                                                                                           ,
+                                                                                                            Debug_DebugString
+#endif
+                                                                                                           );
+            dependsOn = consolidateArrayDataStreamJob.Schedule(dependsOn);
+
+            AccessController.ReleaseAsync(dependsOn);
+            return dependsOn;
+        }
+
+        //*************************************************************************************************************
+        // JOBS
+        //*************************************************************************************************************
+
+        [BurstCompile]
+        private struct ConsolidateArrayDataStreamJob : IJob
```

Comment:
Not necessary now but I feel like there should just be CopyAsync methods on `UnsafeTypedStream and/or `DeferredNativeArray` so we don't need these generic jobs that do nothing unique beyond copy.

Reply:
That's a good point. Will add an issue.